clients/cli/src/prover.rs

```diff
@@ -254,7 +417,22 @@ async fn main() {
                     steps_proven,
                 })),
             };
-            let progress_duration = SystemTime::now().duration_since(progress_time).unwrap();
+            let progress_duration = match SystemTime::now().duration_since(progress_time) {
+                Ok(duration) => duration,
+                Err(e) => {
+                    track(
+                        "time_error".into(),
+                        "Error calculating progress duration".into(),
+                        &ws_addr_string,
+                        json!({
+                            "prover_id": &prover_id,
+                            "error": e.to_string(),
+                        }),
+                    );
+                    // Use a fallback duration or return error
+                    return Err("Failed to calculate progress duration: clock may have gone backwards".into());
```

Comment:
Would it make sense to use `Instant` so that it can't go backwards? See https://linear.app/nexus-labs/issue/NET-224/avoid-systemtime-in-cli

We can do it in a subsequent PR if you prefer.

Reply:
yes i think `instant` is better. good call.